### PR TITLE
add recreate-ica tx.

### PR DESCRIPTION
**Pull request by puneet2019**
## 1. Overview
 adds txn for recreating closed ica channels

## 4. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

File: x/lscosmos/keeper/msg_server.go [hunk 1]
```diff
@@ -541,3 +541,35 @@ func (m msgServer) JumpStart(goCtx context.Context, msg *types.MsgJumpStart) (*t
...
+		}
+	}
+	_, ok = m.icaControllerKeeper.GetOpenActiveChannel(ctx, hostChainParams.ConnectionID, hostAccounts.RewardsAccountPortID())
+	if !ok {
+		err := m.icaControllerKeeper.RegisterInterchainAccount(ctx, hostChainParams.ConnectionID, hostAccounts.RewardsAccountOwnerID)
+		if err != nil {
+			return nil, sdkerrors.Wrap(err, "Could not register ica reward Address")
+		}
+	}
+
```
Comment (arhamchordia) [inline x/lscosmos/keeper/msg_server.go:572]: Emit event missing.

File: x/lscosmos/keeper/msg_server.go [hunk 2]
```diff
@@ -541,3 +541,35 @@ func (m msgServer) JumpStart(goCtx context.Context, msg *types.MsgJumpStart) (*t
 
 	return &types.MsgJumpStartResponse{}, nil
 }
+
+// RecreateICA defines a method for recreating closed ica channels
+func (m msgServer) RecreateICA(goCtx context.Context, msg *types.MsgRecreateICA) (*types.MsgRecreateICAResponse, error) {
+	ctx := sdktypes.UnwrapSDKContext(goCtx)
```
Comment (arhamchordia) [inline x/lscosmos/keeper/msg_server.go:547]: Can anyone do this transaction or only the fee address can do it?

**Review by arhamchordia — DISMISSED** [review]

File: x/lscosmos/keeper/msg_server.go (hunk 2 above)
Comment (puneet2019) [reply x/lscosmos/keeper/msg_server.go:547]: anyone

File: x/lscosmos/keeper/msg_server.go [hunk 3]
```diff
@@ -541,3 +541,51 @@ func (m msgServer) JumpStart(goCtx context.Context, msg *types.MsgJumpStart) (*t
...
+	// sanity check for the arguments of message
+	if ctx.IsZero() {
```
Comment (Anmol1696) [inline x/lscosmos/keeper/msg_server.go:549]: hmm is this correct? `context.Context` we check if `isZero`

**Review by arhamchordia — APPROVED** [review]

File: x/lscosmos/keeper/msg_server.go (hunk 3 above)
Comment (puneet2019) [reply x/lscosmos/keeper/msg_server.go:549]: i think it is not required now, because it is added in abci, wonder from where it arrived first.

**Pull request merged**